### PR TITLE
sign promoted addons for specific applications, where available

### LINKED ISSUE #15315: support autograph signing for promoted for specific applications
#14863 added support for signing other non-recommend promoted groups, but without considering applications.  So the possible states are `['recommended', 'verified', 'sponsored', 'line']`.  

After talking with @Rob--W and @mixedpuppy we're going to interpret `['recommended', 'verified', 'sponsored', 'line']` as being just for desktop Firefox, and add `['recommended-android', 'verified-android', 'sponsored-android', 'line-android']` to support Fenix.

### PR DESCRIPTION
**Pull request by eviljeff**
fixes #15315

### CODE REVIEW

File: scripts/autograph_travis_test_config.yaml [hunk 1]
```diff
@@ -131,7 +131,7 @@ signers:
         path: "mozilla-recommendation.json"
```
Comment (muffinresearch) [inline scripts/autograph_travis_test_config.yaml:131]: Not part of this change, but is the right path or just used for tests? Autograph has `recommendation.json` in their docs https://github.com/mozilla-services/autograph/blob/515ce3638d1b8a8cab1543baf6f33f2223c12831/signer/xpi/README.rst#id1

File: src/olympia/lib/crypto/tests/test_signing.py [hunk 2]
```diff
@@ -364,7 +357,19 @@ def test_call_signing_promoted_recommended_only(self):
         recommendation_data = _get_recommendation_data(
             self.file_.current_file_path)
         assert recommendation_data['addon_id'] == 'xxxxx'
-        assert recommendation_data['states'] == ['recommended']
+        assert sorted(recommendation_data['states']) == states
```
Comment (muffinresearch) [inline src/olympia/lib/crypto/tests/test_signing.py:360]: Could this use sets so order doesn't matter, or should `states` be sorted too?

**Review by muffinresearch — APPROVED** [review]
r+wc

File: scripts/autograph_travis_test_config.yaml (hunk 1 above)
Comment (eviljeff) [reply scripts/autograph_travis_test_config.yaml:131]: looks like their readme is wrong
https://github.com/mozilla-services/autograph/blob/master/autograph.yaml#L987

(and yeah, it's just for tests and local docker dev)

File: src/olympia/lib/crypto/tests/test_signing.py (hunk 2 above)
Comment (eviljeff) [reply src/olympia/lib/crypto/tests/test_signing.py:360]: Casting the JSON into a set would lose the case where a state was included twice in the list, and we don't want that.  States could be sorted too, though it's a kwarg we pass in inside the tests so it's just a matter of remembering to define the kwarg value in alphabetical order or the test fails (we only pass a list of more than 1 item in one place at the moment).
